### ISSUE TITLE
Style/DateTime wrongly triggered on classes `...::DateTime`

### ISSUE THREAD

**Issue by doits** [bbatsov/rubocop#4896]
I've a file with the following line:

    Icalendar::Values::DateTime.new(start_at)

It gives me the following error:

    Style/DateTime: Prefer Date or Time over DateTime.

In this case it is wrong, because `Icalendar::Values::Time` is totally different from `Icalendar::Values::DateTime` (first does not include the date at all).

I think the linter should only be triggered on `/\b(::)?DateTime\b/` and ignore other things. Or am I missing something?

--------

## Expected behavior

`Icalendar::Values::DateTime.new(start_at)` should not trigger `Style/DateTime`

## Actual behavior

It trigger `Style/DateTime`

## Steps to reproduce the problem

Lint a file with `Icalendar::Values::DateTime.new(start_at)`.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.2 x86_64-darwin17)
```

**Comment by Drenmi** [comment]
Yep. There's no reason we should register an offense on a qualified `DateTime` in another namespace. 🙂
